mir-ci/mir_ci/cgroups.py

```diff
@@ -0,0 +1,77 @@
+import os
+from typing import Iterator, Callable, TYPE_CHECKING
+import pathlib
+import logging
+import asyncio
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    CreateReturnType = asyncio.Task["Cgroup"]
+else:
+    CreateReturnType = asyncio.Task
+
+class Cgroup:
+    def __init__(self, pid: int, path: pathlib.Path) -> None:
+        self.pid = pid
+        self.path = path
+
+    @staticmethod
+    def create(pid: int) -> CreateReturnType: # type: ignore
+        async def inner():
+            path = await Cgroup.get_cgroup_dir(pid)
+            return Cgroup(pid, path)
+        task = asyncio.create_task(inner())
+        return task
+
+    @staticmethod
+    async def get_cgroup_dir(pid: int) -> pathlib.Path:
+        MAX_ATTEMPTS = 10
+        parent_path = Cgroup._get_cgroup_dir_internal(os.getpid())
+        path = Cgroup._get_cgroup_dir_internal(pid)
+
+        attempt = 0
+        while path == parent_path:
+            await asyncio.sleep(0.1)
+            path = Cgroup._get_cgroup_dir_internal(pid)
+            attempt = attempt + 1
+            if attempt == MAX_ATTEMPTS:
+                raise RuntimeError(f"Unable to read cgroup directory for pid: {pid}")
+
+        return path
```

Comment:
Here's a Pythonic way :)
```suggestion
        for attempt in range(MAX_ATTEMPTS):
            await asyncio.sleep(0.1)
            path = Cgroup._get_cgroup_dir_internal(pid)
            if path != parent_path:
                return path
        else:
            raise RuntimeError(f"Unable to read cgroup directory for pid: {pid}")
```